### PR TITLE
feat: config param to set when partition is cold

### LINKED ISSUE #6038: Add config param for when to do cold compaction on partition
# Rationale

Currently we hard code 8 as the num of hours without new writes before we start doing cold compaction on that partition. Let us make it config param so we can change it easily.

Thought: do we also want to create config params for when we do hot compaction? Currently we hard code within 4 hours and then 24 hours

### PR DESCRIPTION
**Pull request by NGA-TRAN**
Closes #6038 

- [ ] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: compactor/src/compact.rs [hunk 1]
```diff
@@ -776,7 +778,9 @@ pub mod tests {
 
         // Create a compactor
         let time_provider = Arc::new(SystemProvider::new());
-        let config = make_compactor_config();
+        let mut config = make_compactor_config();
+        // 8 hours to be cold
+        config.minutes_without_new_writes_to_be_cold = 8 * 60;
```
Comment (NGA-TRAN) [inline compactor/src/compact.rs:783]: This was the test for it. I need to make it 8 hours (the hard code before this PR) to keep the tests the same

**Review by domodwyer — APPROVED** [review]

File: clap_blocks/src/compactor.rs [hunk 2]
```diff
@@ -166,6 +166,15 @@ macro_rules! gen_compactor_config {
                 action
             )]
             pub max_num_compacting_files: usize,
+
+            /// Nmmber of minutes without a write to a partition before it is considered cold
```
Comment (alamb) [inline clap_blocks/src/compactor.rs:170]:
```suggestion
            /// Number of minutes without a write to a partition before it is considered cold
            /// and thus a candidate for compaction
```

File: clap_blocks/src/compactor.rs [hunk 3]
```diff
@@ -166,6 +166,15 @@ macro_rules! gen_compactor_config {
...
+            #[clap(
+                long = "compaction-mimutes-without-new-writes-to-be-cold",
```
Comment (alamb) [inline clap_blocks/src/compactor.rs:172]:
```suggestion
                long = "compaction-minutes-without-new-writes-to-be-cold",
```

File: clap_blocks/src/compactor.rs [hunk 4]
```diff
@@ -166,6 +166,15 @@ macro_rules! gen_compactor_config {
...
+                env = "INFLUXDB_IOX_COMPACTION_MINUTE_WITHOUT_NEW_WRITE_TO_BE_COLD",
```
Comment (alamb) [inline clap_blocks/src/compactor.rs:173]:
```suggestion
                env = "INFLUXDB_IOX_COMPACTION_MINUTES_WITHOUT_NEW_WRITE_TO_BE_COLD",
```

File: compactor/src/compact.rs [hunk 5]
```diff
@@ -275,14 +275,15 @@ impl Compactor {
                 ("partition_type", compaction_type.into()),
             ]);
 
-            let time_8_hours_ago = Timestamp::from(self.time_provider.hours_ago(8));
```
Comment (alamb) [inline compactor/src/compact.rs:278]: 👍

File: clap_blocks/src/compactor.rs [hunk 6]
```diff
@@ -166,6 +166,15 @@ macro_rules! gen_compactor_config {
...
+                default_value = "4 * 60",
```
Comment (alamb) [inline clap_blocks/src/compactor.rs:174]: I wonder why this the default value is 4 hours and not 8 hours  (what the hard coded value was) ?

**Review by alamb — APPROVED** [review]
I think there are a few typos in the parameter names

Thanks @NGA-TRAN 
cc @adrian-thurston

File: clap_blocks/src/compactor.rs (hunk 6 above)
Comment (NGA-TRAN) [reply clap_blocks/src/compactor.rs:175]: I wanted to make it shorter as it is the desired now but I agree let us make it the same 8 hours and change the settings in IDPE for productions instead

**Comment by NGA-TRAN** [comment]
I am looking into the test failure 
`end_to_end_cases::cli::compact_and_get_remote_partition`

**Pull request merged**